src/main/drivers/pwm_output.c

```diff
@@ -248,7 +256,15 @@ motorDevice_t *motorPwmDevInit(const motorDevConfig_t *motorConfig, uint8_t moto
 
         /* standard PWM outputs */
         // margin of safety is 4 periods when unsynced
-        const unsigned pwmRateHz = useUnsyncedPwm ? motorConfig->motorPwmRate : ceilf(1 / ((sMin + sLen) * 4));
+        //
+        // Castle link requires 5.5ms wait for tick, after a max 2ms
+        // pulse, which implies an absolute maximum rate of about 133
+        // Hz.  The protocol document specifies 50Hz.  This code
+        // requires at least 9ms + epsilon cycle time, so we set a
+        // maximum of 100Hz to be reasonably safe.
+        const unsigned pwmRateHz = useUnsyncedPwm ?
+                                   ((motorConfig->motorPwmProtocol == PWM_TYPE_CASTLE_LINK) ? MIN(CASTLE_PWM_HZ_MAX, motorConfig->motorPwmRate) : motorConfig->motorPwmRate) :
+                                   ceilf(1 / ((sMin + sLen) * 4));
```

Comment:
Rather than using a different value here without telling the user, there should be a parameter check earlier for the value of motorPwmRate. If it is too high for Castle, it should be changed to a suitable value.
Then the value would be visible in the config too.

Reply:
I had added a check here:

https://github.com/mtrussotto/rotorflight-firmware/blob/castle_telemetry/src/main/config/config.c#L229

Is that sufficient?  I have removed the code in this file.

Reply:
Yes that's good